runtime/profile/_/mod/core/data/dialog/unique/icolle.lua

```diff
@@ -0,0 +1,123 @@
+local Chara = Elona.require("Chara")
+local GUI = Elona.require("GUI")
+local Internal = Elona.require("Internal")
+local Item = Elona.require("Item")
+
+local common = require("data/dialog/common")
+
+local function give_monster_balls()
+   local flag = Internal.get_quest_flag("ambitious_scientist")
+   local found = false
+
+   for _, item in Item.iter(0, 200) do
+      if flag >= 6 then
+         break
+      end
+
+      if item.number > 0 and item.id == "core.monster_ball" and item.subname ~= 0 then
+         found = true
+
+         while item.number > 0 and flag < 6 do
```

Comment:
`GUI.txt(I18N.get("core.locale.talk.unique.icolle.quest.give.deliver", item))` should be inserted.